lib/surface/compiler/eex_engine.ex

```diff
@@ -722,10 +724,10 @@ defmodule Surface.Compiler.EExEngine do
     expr
   end
 
-  defp require_expr(module) do
+  defp require_expr(module, line \\ nil) do
     %AST.Expr{
       value:
-        quote generated: true do
+        quote generated: true, line: line do
```

Comment:
Without this, the error points to the `~H"""`, not to where the component is used.